nemo/utils/exp_manager.py

```diff
@@ -939,38 +939,43 @@ def nemo_topk_check_previous_run(self):
         self.best_model_score = self.best_k_models[self.best_model_path]
 
     def on_save_checkpoint(self, trainer, pl_module, checkpoint):
-        # output = None
         output = super().on_save_checkpoint(trainer, pl_module, checkpoint)
         if not self.always_save_nemo:
             return output
+        # Load the best model and then re-save it
+        app_state = AppState()
+        if app_state.model_parallel_size is not None and app_state.model_parallel_size > 1:
+            logging.warning(f'always_save_nemo will slow down training for model_parallel > 1.')
+        # since we are creating tarfile artifacts we need to update .nemo path
+        app_state.model_restore_path = os.path.abspath(
+            os.path.expanduser(os.path.join(self.dirpath, self.prefix + self.postfix))
+        )
+        injected_best_model_path = inject_model_parallel_rank(self.best_model_path)
```

Comment:
Inject rank only if model parallel is enabled.

Reply:
I think it's a no op if model parallel is not enabled